app/src/main/kotlin/com/sergeylappo/booxrapiddraw/OverlayShowingService.kt

```diff
@@ -50,84 +54,69 @@ class OverlayShowingService : Service() {
 
     override fun onBind(intent: Intent) = null
 
-    //    TODO won't this cause a leak?!
-    class ClickToStopServiceGestureDetector(private val context: Context) : GestureDetector.SimpleOnGestureListener() {
-        override fun onSingleTapConfirmed(e: MotionEvent): Boolean {
-            context.stopService(Intent(context, OverlayShowingService::class.java))
-            exitProcess(0)
-        }
-    }
-
     override fun onCreate() {
         super.onCreate()
-        val channel = NotificationChannel(
-            CHANNEL_ID, "Boox Rapid draw overlay service", NotificationManager.IMPORTANCE_DEFAULT
-        )
 
-        (getSystemService(NOTIFICATION_SERVICE) as NotificationManager).createNotificationChannel(channel)
-
-        val notification = NotificationCompat.Builder(this, CHANNEL_ID)
-            .setContentTitle(getString(R.string.overlay_service_notification_content_title))
-            .setContentText(getString(R.string.overlay_service_notification_content))
-            .setSmallIcon(R.mipmap.ic_launcher_round)
-            .build()
-
-        //noinspection InlinedApi (Seems to work, IDK why, maybe older Android versions might not support this)
-        ServiceCompat.startForeground(this, 1, notification, FOREGROUND_SERVICE_TYPE_SPECIAL_USE)
+        createForegroundNotification()
 
         wm = getSystemService(WINDOW_SERVICE) as WindowManager
 
-        createExitButton()
         createOverlayPaintingView()
 
         initPaint()
         initSurfaceView()
     }
 
-    //    TODO fix suppress
-    @SuppressLint("ClickableViewAccessibility")
-    private fun createExitButton() {
-        val buttonParams = WindowManager.LayoutParams(
-            WindowManager.LayoutParams.WRAP_CONTENT,
-            WindowManager.LayoutParams.WRAP_CONTENT,
-            WindowManager.LayoutParams.TYPE_APPLICATION_OVERLAY,
-            WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE or WindowManager.LayoutParams.FLAG_NOT_TOUCH_MODAL,
-            PixelFormat.TRANSPARENT
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        if (intent?.action == "STOP") {
+            stopSelf()
+            exitProcess(0)
+        }
+
+        val prefs = getSharedPreferences(PREFS_NAME, MODE_PRIVATE)
+        val isRunning = prefs.getBoolean(KEY_IS_RUNNING, false)
+
+        if (isRunning) {
+            Toast.makeText(this, "Terminating Rapid Draw Service...", Toast.LENGTH_SHORT).show()
+            stopSelf()
+            return START_NOT_STICKY // Prevents service from being recreated
+        }
```

Comment:
This causes a crash if the service was running and was closed through the APP's icon:
```
android.app.RemoteServiceException$ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{de47706 u0 com.sergeylappo.booxrapiddraw/.OverlayShowingService}
```